internal/serialization/compact_it_test.go

```diff
@@ -0,0 +1,131 @@
+		{name: "ClusterRestart", f: clusterRestartTest},
+		{name: "JoinedMemberQuery", f: joinedMemberQueryTest},
+		// TODO: add testEntryProcessor when generic record is supported
+		// TODO: add testSchemaReplication when generic record is supported
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			tc.f(t)
+		})
+	}
+}
+
+func compactBasicTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testBasic
+	tcx := it.CompactTestContext{
+		T:            t,
+		Serializers1: []serialization.CompactSerializer{EmployeeDTOCompactSerializer{}},
+		Serializers2: []serialization.CompactSerializer{EmployeeDTOCompactSerializerV2{}},
+	}
+	tcx.Tester(func(tcx it.CompactTestContext) {
+		ctx := context.Background()
+		v := EmployeeDTO{age: 30, id: 102310312}
+		map1 := it.MustValue(tcx.Client1.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		it.Must(map1.Set(ctx, 1, v))
+		map2 := it.MustValue(tcx.Client2.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		it.AssertEquals(t, v, it.MustValue(map2.Get(ctx, 1)))
+	})
+}
+
+func basicQueryTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testBasicQuery
+	tcx := it.CompactTestContext{
+		T:            t,
+		Serializers1: []serialization.CompactSerializer{EmployeeDTOCompactSerializer{}},
+		Serializers2: []serialization.CompactSerializer{EmployeeDTOCompactSerializerV2{}},
+	}
+	tcx.Tester(func(tcx it.CompactTestContext) {
+		ctx := context.Background()
+		map1 := it.MustValue(tcx.Client1.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		for i := int32(0); i < 100; i++ {
+			it.MustValue(map1.Put(ctx, i, EmployeeDTO{age: i, id: 102310312}))
+		}
+		map2 := it.MustValue(tcx.Client2.GetMap(ctx, tcx.MapName)).(*hz.Map)
+		ks := it.MustValue(map2.GetKeySetWithPredicate(ctx, predicate.SQL("age > 19"))).([]interface{})
+		assert.Equal(t, 80, len(ks))
+	})
+}
+
+func joinedMemberQueryTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testJoinedMemberQuery
+	tcx := it.CompactTestContext{
+		T:            t,
+		Serializers1: []serialization.CompactSerializer{EmployeeDTOCompactSerializer{}},
+	}
+	tcx.Tester(func(tcx it.CompactTestContext) {
+		ctx := context.Background()
+		for i := int32(0); i < 100; i++ {
+			it.MustValue(tcx.M.Put(ctx, i, EmployeeDTO{age: i, id: 102310312}))
+		}
+		it.MustValue(tcx.Cluster.StartMember(ctx))
+		err := tcx.ExecuteScript(ctx, fmt.Sprintf(`
+			var m = instance_2.getMap("%s");
+			var size = m.keySet(com.hazelcast.query.Predicates.sql("age > 19")).size();
+			if (size != 80) {
+				throw new Error("size is " + size);
+			}			
+		`, tcx.MapName))
+		assert.NoError(t, err)
+	})
+}
+
+func clusterRestartTest(t *testing.T) {
+	// ported from: com.hazelcast.internal.serialization.impl.compact.integration.CompactFormatIntegrationTest#testJoinedMemberQuery
```

Comment:
this is the same as above, probably miscopy of the original test from above

Reply:
Replaced with the correct ref here: https://github.com/hazelcast/hazelcast-go-client/pull/942/commits/62dd8899f77fe3fbee9e3e258f319335907831c7